packages/interface-ipfs-core/src/add.js

```diff
@@ -499,12 +502,12 @@ module.exports = (common, options) => {
 
     it('should add from a URL with wrap-with-directory=true', async () => {
       const filename = `TEST${Date.now()}.txt` // also acts as data
-      const url = echoUrl(filename) + '?foo=bar#buzz'
+      const url = echoUrl(filename)
       const addOpts = { wrapWithDirectory: true }
 
       const [result, expectedResult] = await Promise.all([
         all(ipfs.add(urlSource(url), addOpts)),
-        all(ipfs.add([{ path: filename, content: Buffer.from(filename) }], addOpts))
+        all(ipfs.add([{ path: 'download', content: Buffer.from(filename) }], addOpts))
```

Comment:
Why has `filename` changed to `'download'`?

Reply:
this has to do with the echoserver download endpoint, the previous server was simpler and one could do just any path and it would respond, but the new has defined endpoints for different stuff.
So urlSource uses the pathname of the url for the path (http://localhost:3000/download?data=helloworld) we get "download" in file path, i could do a catch-all endpoint just for this use-case so we can build the file path with the url pathname but i didn’t see the value because the test still is testing whats needed so i left it like that.